src/components/HexagonGrid.tsx

```diff
@@ -81,11 +86,37 @@ function HexagonGrid(props: HexagonGridProps) {
     return dimensions;
   };
 
-  function testOnClick(hexagon: HexData) {
-    console.log("clicked! now", hexagon);
+  function updateHexagon(hexagon: HexData) {
+    const hex = hexagons.find(hex => hex.index === hexagon.index)
+
+    if (!hex) {
+      const newHexagon: HexData = {
+        index: hexagon.index,
+        terrain: brushSelection.terrain ? brushSelection.terrain : Terrain.NORMAL,
+        bonus: brushSelection.icon ? {icon: brushSelection.icon} : undefined,
+      }
+      hexagons.push(newHexagon)
+    } else {
+      hex.terrain = brushSelection.terrain !== undefined
+        ? brushSelection.terrain 
+        : hex.terrain
+      
+      if (brushSelection.icon !== undefined) {
+        hex.bonus = {
+          ...hex.bonus,
+          icon: brushSelection.icon
+        }
+      }
+
+      if (brushSelection.deleteIcon) {
+        hex.bonus = undefined
+      }
+    }
+    
+    setHexagons([...hexagons]);
   }
 
-  return (<svg width="1000" height="650">
+  return (<svg width="1000" height="750">
     {times(hexInfo.rows, (row) => {
```

Comment:
don't need to do here, but we should think about this in the future: consider extracting this to it's own function to make this more readable